src/index.js

```diff
@@ -0,0 +1,147 @@
+'use strict'
+
+const {
+  UnixFS: { unmarshal: decodeUnixFs }
+} = require('ipfs-unixfs')
+
+const { codecs, blocksTable, primaryKeys, carsTable, publishingQueue } = require('./config')
+const { logger, elapsed } = require('./logging')
+const { openS3Stream } = require('./source')
+const { readDynamoItem, writeDynamoItem, publishToSQS } = require('./storage')
+const { forEach } = require('hwp')
+
+function decodeBlock(block) {
+  const codec = codecs[block.cid.code]
+
+  if (!codec) {
+    logger.error(`Unsupported codec ${block.cid.code} in the block at offset ${block.blockOffset}`)
+    throw new Error(`Unsupported codec ${block.cid.code} in the block at offset ${block.blockOffset}`)
+  }
+
+  // Decoded the block data
+  const data = codec.decode(block.data)
+
+  if (codec.label.startsWith('dag')) {
+    const { type, blocks } = decodeUnixFs(data.Data)
+
+    data.Data = { type, blocks }
+  }
+
+  return { codec: codec.label, data }
+}
+
+async function storeNewBlock(car, type, block, data = {}) {
+  for (const link of data?.Links ?? []) {
+    link.Hash = link.Hash.toString()
+  }
+
+  const cid = block.cid.toString()
+
+  await writeDynamoItem(true, blocksTable, primaryKeys.blocks, cid, {
+    type,
+    createdAt: new Date().toISOString(),
+    cars: [{ car, offset: block.blockOffset, length: block.blockLength }],
+    data
+  })
+
+  return publishToSQS(publishingQueue, cid)
+}
+
+async function appendCarToBlock(block, cars, carId) {
+  cars.push({ car: carId, offset: block.blockOffset, length: block.blockLength })
+
+  return writeDynamoItem(false, blocksTable, primaryKeys.blocks, block.cid.toString(), {
+    cars: cars.sort((a, b) => {
+      return a.offset !== b.offset ? a.offset - b.offset : a.car.localeCompare(b.car)
+    })
+  })
+}
+
+async function main(event) {
+  const start = process.hrtime.bigint()
+
+  // For each Record in the event
+  let currentCar = 0
+  const totalCars = event.Records.length
+
+  for (const record of event.Records) {
```

Comment:
Do we need this for loop? 
This function will be triggered whenever there is a new file and we only care about the event.Records[0]. Right?

Reply:
I agree with you in theory.
In practice, since it is an array, I just want to make we're ready if AWS changes behavior. It won't hurt anyway :)